core/src/main/java/digitalusage/rogers/com/actors/util/http/TopUpPurchasedClient.java

```
@@ -69,8 +69,12 @@ public TopUpPurchasedClient(String url, String table, long ttl, String ucsClient
 
     public static final Function<RetrievePurchasedTopUpsRequest, JsonObject> maptoPurchasedRequest = request -> {
 
+        String genesisApplicationId = request.getContext().getApplicationId();
+        String ucsApplicationId = ucsApplicationIdMapping.getOrDefault(genesisApplicationId,ucsApplicationIdMapping.get ("UteMyRogersApp"));
```

I think it might be a better idea here to throw an exception and return an error code for invalid application Id instead of falling back to using this one. Thoughts?

---

For now it does not seem like UCS is doing anything with ApplicationId. I will discuss this one with Aowss

---

i think the application id should drive the client id, the channel and the brand.
I think it should be checked in the controller and throw an error there.
Having the value here is probably not a good idea.
We need to streamline this.

---

Agreed. In fact, I was going to suggest earlier that we should make this more functional and not initialize this class with the mapping — instead have it passed in. Would make it easier to test as it would be decoupled with the configuration.

---

Done